dataflow/src/main/java/org/checkerframework/dataflow/busyexpression/BusyExprTransfer.java

```diff
@@ -0,0 +1,113 @@
+package org.checkerframework.dataflow.busyexpression;
+
+import org.checkerframework.dataflow.analysis.BackwardTransferFunction;
+import org.checkerframework.dataflow.analysis.RegularTransferResult;
+import org.checkerframework.dataflow.analysis.TransferInput;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.UnderlyingAST;
+import org.checkerframework.dataflow.cfg.node.AbstractNodeVisitor;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.ObjectCreationNode;
+import org.checkerframework.dataflow.cfg.node.ReturnNode;
+import org.checkerframework.dataflow.cfg.node.StringConcatenateAssignmentNode;
+
+import java.util.List;
+
+/** A busy expression transfer function */
+public class BusyExprTransfer
+        extends AbstractNodeVisitor<
+                TransferResult<BusyExprValue, BusyExprStore>,
+                TransferInput<BusyExprValue, BusyExprStore>>
+        implements BackwardTransferFunction<BusyExprValue, BusyExprStore> {
+
+    @Override
+    public BusyExprStore initialNormalExitStore(
+            UnderlyingAST underlyingAST, List<ReturnNode> returnNodes) {
+        return new BusyExprStore();
+    }
+
+    @Override
+    public BusyExprStore initialExceptionalExitStore(UnderlyingAST underlyingAST) {
+        return new BusyExprStore();
+    }
+
+    @Override
+    public RegularTransferResult<BusyExprValue, BusyExprStore> visitNode(
+            Node n, TransferInput<BusyExprValue, BusyExprStore> p) {
+        return new RegularTransferResult<>(null, p.getRegularStore());
+    }
+
+    @Override
+    public RegularTransferResult<BusyExprValue, BusyExprStore> visitAssignment(
+            AssignmentNode n, TransferInput<BusyExprValue, BusyExprStore> p) {
+        RegularTransferResult<BusyExprValue, BusyExprStore> transferResult =
+                (RegularTransferResult<BusyExprValue, BusyExprStore>) super.visitAssignment(n, p);
+        processBusyExprInAssignment(
+                n.getTarget(), n.getExpression(), transferResult.getRegularStore());
+        return transferResult;
+    }
+
+    @Override
+    public RegularTransferResult<BusyExprValue, BusyExprStore> visitStringConcatenateAssignment(
```

Comment:
Note: I think Alex is going to remove this method.

Reply:
Ok, thanks, let me remove that